mlflow/tracking/client.py

```diff
@@ -233,14 +234,19 @@ def log_batch(self, run_id, metrics=(), params=(), tags=()):
 
     def log_artifact(self, run_id, local_path, artifact_path=None):
         """
-        Write a local file to the remote ``artifact_uri``.
+        Write a local file or directory to the remote ``artifact_uri``.
 
-        :param local_path: Path to the file to write.
+        :param local_path: Path to the file or directory to write.
         :param artifact_path: If provided, the directory in ``artifact_uri`` to write to.
         """
         run = self.get_run(run_id)
         artifact_repo = get_artifact_repository(run.info.artifact_uri)
-        artifact_repo.log_artifact(local_path, artifact_path)
+        if os.path.isdir(local_path):
+            dir_name = os.path.basename(os.path.normpath(local_path)) \
+                if artifact_path is None else artifact_path
```

Comment:
Hm we should actually decide what makes more sense here. If we pass `artifact_path` directory to `log_artifacts`, I think we'll log the directory's contents under `artifact_path`, instead of under `artifact_path + "/" + directory_name` (which is what happens for files passed to `log_artifact`)

Reply:
Implemented this and manually verified it works.